go/cmd/dolt/commands/dump.go

```diff
@@ -0,0 +1,324 @@
+		"[-f] [-r {{.LessThan}}result-format{{.GreaterThan}}] ",
+	},
+}
+
+type dumpOptions struct {
+	format string
+	dest   mvdata.DataLocation
+}
+
+type tableOptions struct {
+	tableName  string
+	src        mvdata.TableDataLocation
+	dest       mvdata.DataLocation
+	srcOptions interface{}
+}
+
+func (m tableOptions) WritesToTable() bool {
+	return false
+}
+
+func (m tableOptions) SrcName() string {
+	return m.src.Name
+}
+
+func (m tableOptions) DestName() string {
+	if t, tblDest := m.dest.(mvdata.TableDataLocation); tblDest {
+		return t.Name
+	}
+	if f, fileDest := m.dest.(mvdata.FileDataLocation); fileDest {
+		return f.Path
+	}
+	return m.dest.String()
+}
+
+func (m dumpOptions) DumpDestName() string {
+	if t, tblDest := m.dest.(mvdata.TableDataLocation); tblDest {
+		return t.Name
+	}
+	if f, fileDest := m.dest.(mvdata.FileDataLocation); fileDest {
+		return f.Path
+	}
+	return m.dest.String()
+}
+
+func checkOverwrite(ctx context.Context, root *doltdb.RootValue, fs filesys.ReadableFS, force bool, dest mvdata.DataLocation) (bool, error) {
+	if _, isStream := dest.(mvdata.StreamDataLocation); isStream {
+		return false, nil
+	}
+	if !force {
+		return dest.Exists(ctx, root, fs)
+	}
+	return false, nil
+}
+
+// getExportDestination returns an export destination corresponding to the input parameters
+func getExportDestination(path string) mvdata.DataLocation {
+
+	destLoc := mvdata.NewDataLocation(path, "")
+
+	switch val := destLoc.(type) {
+	case mvdata.FileDataLocation:
+		if val.Format == mvdata.InvalidDataFormat {
+			cli.PrintErrln(
+				color.RedString("Could not infer type file '%s'\n", path),
+				"File extensions should match supported file types, or should be explicitly defined via the file-type parameter")
+			return nil
+		}
+
+	case mvdata.StreamDataLocation:
+		if val.Format == mvdata.InvalidDataFormat {
+			val = mvdata.StreamDataLocation{Format: mvdata.CsvFile, Reader: os.Stdin, Writer: iohelp.NopWrCloser(cli.CliOut)}
+			destLoc = val
+		} else if val.Format != mvdata.CsvFile && val.Format != mvdata.PsvFile {
+			cli.PrintErrln(color.RedString("Cannot export this format to stdout"))
+			return nil
+		}
+	}
+
+	return destLoc
+}
+
+func getDumpArgs(apr *argparser.ArgParseResults) (*dumpOptions, errhand.VerboseError) {
+
+	if apr.NArg() > 0 {
+		return nil, errhand.BuildDError("too many arguments").SetPrintUsage().Build()
+	}
+
+	var fileName string
+	resultFormat, _ := apr.GetValue(FormatFlag)
+
+	switch resultFormat {
+	case "", "sql", ".sql":
+		fileName = "doltdump.sql"
+	case "csv", ".csv":
+		//handle CSV filetype
+		//maybe create dir 'doltdump' and put all the csv dump files into it
+	default:
+		return nil, errhand.BuildDError("invalid result format").SetPrintUsage().Build()
+	}
+
+	fileLoc := getExportDestination(fileName)
+
+	return &dumpOptions{
+		format: resultFormat,
+		dest:   fileLoc,
+	}, nil
+}
+
+func getTableArgs(tblName string, destination mvdata.DataLocation) (*tableOptions, errhand.VerboseError) {
```

Comment:
Usually call methods like this new*, so newTableOpts

Also the second return param is always nil, so remove it